### DISCUSSION TITLE
Cannot handle multiple requests on same http path

### DISCUSSION THREAD

**Discussion by LaurentBERTHOLLE** [nodejs/help#4745] [Q&A] [answered]
### Version

v16.19.1

### Platform

Darwin laurenzos-macbook-pro.home 22.5.0 Darwin Kernel Version 22.5.0: Thu Jun  8 22:22:22 PDT 2023; root:xnu-8796.121.3~7/RELEASE_X86_64 x86_64

### What steps will reproduce the bug?

Copy and paste the following code in the index.js file
```js
const http = require('http');

const hostname = '127.0.0.1';
const port = 3000;

const server = http.createServer((req, res) => {
  const p = new Promise((resolve) => {
    setTimeout(() => {
      resolve()
    }, 10000)
  })
  p.then(() => {
    res.statusCode = 200;
    res.setHeader('Content-Type', 'text/plain');
    res.end("hello " + req.url);
  })
});

server.listen(port, hostname, () => {
  console.log(`Server running at http://${hostname}:${port}/`);
});
```
and start the server.

First case:
When you launch 2 tabs on your browser with different paths such as /foo and /bar, the two tabs are loading for 5 seconds and then display respectively "hello /foo" and "hello bar" at the same time.
https://watch.screencastify.com/v/ST2EyvSW4vMwyCZZ7xAU

Second case:
When you launch 2 tabs on your browser with the same path (/foo for instance), the first tab is loading for 5 seconds and then displays "hello /foo" and the second tab is loading for 10 seconds and display "hello /foo".
https://watch.screencastify.com/v/FqB1wEobYXkBGjbIvDbj

### How often does it reproduce? Is there a required condition?

Always reproducible.

### What is the expected behavior? Why is that the expected behavior?

I would expect the second case to work like the first case:
when launching 2 tabs on your browser with the same path (/foo for instance), the two tabs are loading for 5 seconds and then display "hello /foo" at the same time.

### What do you see instead?

In the second case, the second tab is loading for 10 seconds, as if the second request was beginning to be handled only after the first one is finished, while in the first case, they seem to be handled at the same time.

### Additional information

I know Node.js is single-threaded, however it does not explain the difference between the two cases.

**Comment by tniessen** [comment]
I can't reproduce this. Could you please use `curl` or so to take timing measurements to rule out the usual browser oddities?

**Comment by bnoordhuis** [comment, marked as answer]
Converted to a discussion because there's no bug. Your question really is "why does my browser work this way?" and the answer is "because it's configured to work like that."

Like Tobias said, use curl instead.
